chore: add abandoned state (#11)
electron/rfcs: README.md

  **An Active RFC means that the idea is worth being implemented and explored, not that the
  maintainers are committing to adding it to Electron.** If a new development causes an Active
  RFC to become unnecessary in any way, maintainers can choose to ultimately decline it.

> - **Abandoned:** An RFC is considered Abandoned if it has no owner actively pushing it forward.
>   It may be reopened in the same PR or as a new PR if an owner renews interest in pursuing it.

Reviewer: nit: clarify what happens when an RFC is abandoned or reopened

```suggestion
- **Abandoned:** An RFC will be closed as Abandoned if it has no owner actively pushing it forward.
  The original owner may reopen the PR, or a new owner may open a new PR if they are interested in pursuing it.
```
